### DISCUSSION TITLE
Parameter to rename the encrypted and decrypted files.

### DISCUSSION THREAD

**Discussion by bogronaso** [samuel-lucas6/Kryptor#75] [Ideas]
Dear Kryptor Team,

I’d like to suggest adding a parameter that allows users to rename files during the encryption and decryption processes. This feature would enhance usability and improve file management for users.

Thank you for considering this enhancement!

Best,
Bogronaso.

**Comment by samuel-lucas6** [comment]
Hi, thanks for the feature request. The reason this isn't supported is because you can specify multiple files/directories at once, which makes this messy. I agree it could be useful for single files, but I don't really want to add an option that only works when one file is specified.

I'd like to make another file encryption tool in the future that's as simple as possible, kind of like [Encpipe](https://github.com/jedisct1/encpipe). That would allow you to specify the name of the output file.

**Reply by hakavlad** [reply]
>simple as possible, kind of like [Encpipe](https://github.com/jedisct1/encpipe)

What features will this tool have, if it's not a secret? How will encpipe be different?

**Reply by samuel-lucas6** [reply]
I haven't looked at Encpipe properly, and I have no idea how far off this will be, but I was thinking just password-based encryption with Argon2id, AEGIS-256, a larger chunk size than Kryptor, padding, and maybe YubiKey support like with [Cahir](https://github.com/samuel-lucas6/Cahir). So, it would be more performant and have a few extra features over Encpipe.

The incentive is to provide a blueprint for how password-based file encryption should be done because that's likely what most people are using, not hybrid encryption. For personal use cases, hybrid encryption is unnecessary. Something simple means it's auditable and easier to duplicate.

It might also allow experimenting with some ideas for Kryptor, although the hybrid encryption and signatures are what make Kryptor more complicated, especially going forward with the post-quantum algorithms. That's going to completely screw up the UX and affect the security properties.

**Reply by hakavlad** [reply]
Thanks for the answer.

>password-based file encryption should be done because that's likely what most people are using, not hybrid encryption

Also most people prefer GUI. In that case, will it have significant advantages over Picocrypt and age (besides the PURB format and Yubikey support)? BTW Picocrypt was [audited](https://github.com/Picocrypt/storage/blob/main/Picocrypt.Audit.Report.pdf).

See also: https://github.com/70sh1/eddy

**Reply by samuel-lucas6** [reply]
A GUI is significantly more complicated if you want something cross-platform. I don't think MAUI works with Linux. It would also interfere with the security of the password entry. But it would be good experience.

The main advantages would probably be simplicity and performance. The goal wouldn't be to have every feature under the sun or multiple cipher suites, etc. It would deliberately be barebones and probably have no new/requested features implemented.

The Picocrypt audit really shows what a waste of money audits/pentests can be. There was nothing that couldn't have been found by auditing the code yourself for free rather than paying several thousand quid. Then it doesn't cover any code changes that are made after the audit, which is how things like [this](https://www.daemonology.net/blog/2011-01-18-tarsnap-critical-security-bug.html) happen.

I don't know anything about eddy, but it appears to be authenticating the entire ciphertext in one go, has an unnecessarily large tag, uses scrypt rather than Argon2, and generates the BLAKE2b key from the ChaCha20 keystream for some reason.

**Reply by hakavlad** [reply]
>has an unnecessarily large tag

I saw that you recommended it for additional security:

>A 256-bit authentication tag is sufficient for most use cases: however, a 512-bit tag provides additional security if you’re concerned about quantum computing.

-- Cryptography-Guidelines

Is this still relevant?

**Reply by hakavlad** [reply]
>There was nothing that couldn't have been found by auditing the code yourself for free

It's bs. Picocrypt users are not cryptographers and not gophers. It's also not free, because time=money. "auditing the code yourself for free" needs time, especially for those who are not experts.

**Reply by samuel-lucas6** [reply]
> Is this still relevant?

As I keep saying, the guidelines need an overhaul, but it's a massive job, I'm extremely slow at writing/researching, and it feels like I'm working on too many things at once.

> It's bs. Picocrypt users are not cryptographers and not gophers. It's also not free, because time=money. "auditing the code yourself for free" needs time, especially for those who are not experts.

I'm saying the author should audit the code. If it's a relatively simple program, it's less likely that you'd miss something.

Other people may also voluntarily audit the code for free, or you could try a bug bounty, which would be cheaper but results in loads of spam.

**Reply by hakavlad** [reply]
>I'm saying the author should audit the code.

The author is not an expert. He can't even conduct a qualified audit. An audit is an _independent check_, that's the point.

**Reply by samuel-lucas6** [reply]
> The author is not an expert. He can't even conduct a qualified audit. An audit is an independent check, that's the point.

I'm using the word audit to mean examine, and audits as a concept don't have to be done by a third party.

Obviously, a developer is probably less familiar with identifying/fixing vulnerabilities, but a lot of vulnerabilities don't apply to this type of application, the developer should have a better understanding of what the code is doing, the developer doesn't necessarily have a deadline, and looking at your own code doesn't cost a few thousand each time.

Peer review is never a bad thing, but it's hard to get in practice and not necessarily required.
